src/error.rs

```diff
@@ -1,4 +1,7 @@
-use std::{error, fmt};
+#[cfg(not(feature = "std"))]
+use core as std;
+
+use std::fmt;
```

Comment:
Could we just use `core` directly, without special-casing no_std?

So this would just be `use core::fmt and further below `core::result::Result<T, Error>`.